packages/components/inputs/rich-text-utils/src/rich-text-body/dropdown.tsx

```diff
@@ -10,26 +14,69 @@ import {
   DropdownItem as StyledDropdownItem,
 } from './dropdown.styles';
 
-const getIsSelected = (props, item) =>
+type Props = {
+  label: string;
+  isMulti: boolean;
+  value: string[] | string;
+  isDisabled?: boolean;
+  isReadOnly?: boolean;
+  onChange?: (event?: unknown) => void;
+  components: {
+    Item: FunctionComponent<unknown>;
+    Label: FunctionComponent<unknown>;
+  };
+  options: Array<TItem>;
+};
+
+type TItem = {
+  value: string;
+  label: string;
+};
+
+const getIsSelected = (props: Props, item: TItem) =>
   !props.isMulti
-    ? item.value === props.value
-    : props.value.find((selectedItem) => selectedItem === item.value);
+    ? item.value === (props.value as string)
+    : (props.value as string[]) &&
+      (props.value as string[]).find(
+        (selectedItem) => selectedItem === item.value
+      );
 
 const Label = styled.div;
 
-const Dropdown = (props) => {
-  const DropdownItem = props.components.Item;
+const Dropdown = (props: Props) => {
+  const DropdownItem: FunctionComponent<{
+    value: string | string[];
+    isSelected: unknown;
+  }> = props.components.Item;
   const DropdownLabel = props.components.Label;
-  const isIndeterminate =
-    props.isMulti && props.value && props.value.length > 0;
+  const isIndeterminate: boolean =
+    props.isMulti &&
+    (props.value as unknown[]) &&
+    (props.value as unknown[]).length > 0;
+
+  type THeadings = {
+    label?: string;
+  };
+
+  const Downshift = DownshiftUntyped as ElementType;
 
   return (
     <Downshift
       onChange={props.onChange}
       selectedItem={props.value}
-      itemToString={(headings) => (headings ? headings.label : '')}
+      itemToString={(headings: THeadings) => (headings ? headings?.label : '')}
```

Comment:
nit: Now that we're using the optional chaining, it seems we don't need the ternary anymore:
```suggestion
      itemToString={(headings: THeadings) => (headings?.label || '')}
```